src/main/java/org/opentripplanner/graph_builder/module/StreetLinkerModule.java

```diff
@@ -102,6 +103,31 @@ public void linkTransitStops(Graph graph, TransitModel transitModel) {
       );
     }
 
+    // The stops that are used by transit capable of transporting cars need to be connected to the road network (e.g. car ferries).
+    Set<StopLocation> stopLocationsUsedForCarsAllowedTrips = Set.of();
+    stopLocationsUsedForCarsAllowedTrips =
+      transitModel
+        .getAllTripPatterns()
+        .stream()
+        .filter(t ->
+          t
+            .getScheduledTimetable()
+            .getTripTimes()
+            .stream()
+            .anyMatch(tt -> tt.getTrip().getCarsAllowed() == CarAccess.ALLOWED)
+        )
+        .flatMap(t -> t.getStops().stream())
+        .collect(Collectors.toSet());
+
+    stopLocationsUsedForCarsAllowedTrips.addAll(
+      stopLocationsUsedForCarsAllowedTrips
+        .stream()
+        .filter(GroupStop.class::isInstance)
+        .map(GroupStop.class::cast)
+        .flatMap(g -> g.getChildLocations().stream().filter(RegularStop.class::isInstance))
+        .toList()
+    );
```

Comment:
Can you extract that into a method and convert the line comments to Javadoc?

Reply:
actually, can also extract a method for the flex stops?

Reply:
I created methods for both in the `transitModel`. This will also be useful later when working on the `transferRequest` stuff related to cars on transit